src/services/completions.ts

```diff
@@ -585,9 +586,11 @@ namespace ts.Completions {
             }
 
             const { name, needsConvertPropertyAccess } = info;
+            const sortText = isDeprecated(symbol, typeChecker) ? SortText.Deprecated :
+                symbolToSortTextMap && symbolToSortTextMap[getSymbolId(symbol)] || SortText.LocationPriority;

```

Comment:
Should these be instead: 
DeprecatedLocalDeclarationPriority
        DeprecatedLocationPriority,
        DeprecatedOptionalMember,
        DeprecatedMemberDeclaredBySpreadAssignment,
        DeprecatedSuggestedClassMembers,
        DeprecatedGlobalsOrKeywords,
        DeprecatedAutoImportSuggestions,

Reply:
@sheetalkamat Thanks for the review 👍🏻 . As I understood, the `SortText` need to change like so., Am I right?
```ts
export enum SortText {
   LocalDeclarationPriority = "0",
   DeprecatedLocalDeclarationPriority = "1",
   LocationPriority = "2",
   DeprecatedLocationPriority = "3",
   OptionalMember = "4",
   DeprecatedOptionalMember = "5",
   MemberDeclaredBySpreadAssignment = "6",
   DeprecatedMemberDeclaredBySpreadAssignment = "7",
   SuggestedClassMembers = "8",
   DeprecatedSuggestedClassMembers = "9",
   GlobalsOrKeywords = "10",
   DeprecatedGlobalsOrKeywords = "11",
   AutoImportSuggestions = "12",
   DeprecatedAutoImportSuggestions = "13",
   JavascriptIdentifiers = "14"
}
```

Reply:
I am not sure about ordering... @RyanCavanaugh @DanielRosenwasser @uniqueiniquity @jessetrinity would be able to comment on the ordering but it seems we would want to prioritize say localtion priority deprecated over deprecated auto imports so i am not sure if all deprecated go in the end in the same order as their non deprecated ones or not. Also I think there is some issue with changing number for `JavascriptIdentifiers` that @uniqueiniquity is able to weigh in...

Reply:
The issue for us is changing the value of `GlobalsOrKeywords`, since we add global completions in on the editor side for snippets. So I have a slight preference for adding all the distinct deprecated categories at the end instead of interleaving them from that perspective as well as from a usability perspective. But we certainly can handle the change in the editor if we needed to interleave the new categories for some reason.

Reply:
@uniqueiniquity Thanks for the clarification. We need to add a new property _(DeprecatedGlobalsOrKeywords)_ to `SortText` and only sort the deprecated GlobalsOrKeywords entities. I'm right?
 
```ts
export enum SortText {
   LocalDeclarationPriority = "0",
   LocationPriority = "1",
   OptionalMember = "2",
   MemberDeclaredBySpreadAssignment = "3",
   SuggestedClassMembers = "4",
   GlobalsOrKeywords = "5",
   AutoImportSuggestions = "6",
   JavascriptIdentifiers = "7",
   DeprecatedGlobalsOrKeywords = "8"
}
```

Reply:
I apologize for not being clear. Here's what I think is correct, if others feel that the deprecated items need to be sorted within themselves:
```ts
export enum SortText {
   LocalDeclarationPriority = "0",
   LocationPriority = "1",
   OptionalMember = "2",
   MemberDeclaredBySpreadAssignment = "3",
   SuggestedClassMembers = "4",
   GlobalsOrKeywords = "5",
   AutoImportSuggestions = "6",
   DeprecatedLocalDeclarationPriority = "7",
   DeprecatedLocationPriority = "8",
   DeprecatedOptionalMember = "9",
   DeprecatedMemberDeclaredBySpreadAssignment = "10",
   DeprecatedSuggestedClassMembers = "11",
   DeprecatedGlobalsOrKeywords = "12",
   DeprecatedAutoImportSuggestions = "13",
   JavascriptIdentifiers = "14"
}
```
If the deprecated items do not need to be sorted within themselves, then I am fine with the PR as it currently is.

Reply:
@uniqueiniquity Thanks.

Reply:
Do we need to sort the deprecated elements within ourselves?

/cc @RyanCavanaugh @DanielRosenwasser @jessetrinity @sandersn @sheetalkamat

Reply:
We discussed this today and indeed we want the deprecated entries sorted among themselves.